tock/employees/tests/test_views.py

```diff
@@ -1,30 +1,154 @@
-from unittest.mock import Mock, patch
-
-from django.conf import settings
 from django.test import TestCase, Client
 from django.contrib.auth import get_user_model
 from django.core.urlresolvers import reverse
 
-from employees.views import UserFormView
+import datetime
+from employees.views import parse_date, UserListView
+from employees.models import UserData
+
 
 class UserViewTests(TestCase):
 
     fixtures = ['tock/fixtures/dev_user.json']
-    
+
     def setUp(self):
-        self.regular_user = get_user_model().objects.create(username="regular.user")
+        self.regular_user = get_user_model().objects.create(
+            username="regular.user")
+        UserData(
+            user=self.regular_user,
+            start_date=datetime.datetime(2014, 1, 1),
+            end_date=datetime.datetime(2016, 1, 1)
+        ).save()
+
+    def test_parse_date(self):
+        """ Test that the parse date function returns datetime obj when
+        input is not NA """
+        date = 'NA'
+        self.assertEqual(parse_date(date=date), None)
+        date = "01/01/2013"
+        self.assertEqual(parse_date(date=date), datetime.datetime(2013, 1, 1))
+
+    def test_UserListViewPermissionforAdmin(self):
+        """ Ensure that UserListView works for admin """
+        c = Client(HTTP_X_FORWARDED_EMAIL='test.user@gsa.gov')
```

Comment:
@jmcarp k, I've moved the view tests to WebTest. The form submission feature is great!